tests/components/sensor/test_template.py

```diff
@@ -90,7 +86,7 @@ def test_invalid_name_does_not_create(self):
 
     def test_invalid_sensor_does_not_create(self):
         """Test invalid sensor."""
-        assert sensor.setup(self.hass, {
+        assert not bootstrap.setup_component(self.hass, 'sensor', {
```

Comment:
If I remove the `not` from one of the tests - here is the log.
````
__________________________________________________________________________________________ TestTemplateSensor.test_invalid_name_does_not_create __________________________________________________________________________________________

self = <tests.components.sensor.test_template.TestTemplateSensor object at 0x1030d16d8>

    def test_invalid_name_does_not_create(self):
        """Test invalid name."""
        import pdb; pdb.set_trace()
>       assert bootstrap.setup_component(self.hass, 'sensor', {
            'sensor': {
                'platform': 'template',
                'sensors': {
                    'test INVALID sensor': {
                        'value_template':
                            "{{ states.sensor.test_state.state }}"
                    }
                }
            }
        })
E       assert <function setup_component at 0x10318c488>(<homeassistant.core.HomeAssistant object at 0x1030d17f0>, 'sensor', {'sensor': {'platform': 'template', 'sensors': {'test INVALID sensor': {'value_template': '{{ states.sensor.test_state.state }}'}}}})
E        +  where <function setup_component at 0x10318c488> = bootstrap.setup_component
E        +  and   <homeassistant.core.HomeAssistant object at 0x1030d17f0> = <tests.components.sensor.test_template.TestTemplateSensor object at 0x1030d16d8>.hass

tests/components/sensor/test_template.py:75: AssertionError
-------------------------------------------------------------------------------------------------------------- Captured log --------------------------------------------------------------------------------------------------------------
bootstrap.py               388 ERROR    Invalid config for [sensor.template]: [test INVALID sensor] is an invalid option for [sensor.template]. Check: sensor.template->sensors->test INVALID sensor.
---------------------------------------------------------------------------------------------------------- Captured stderr call ----------------------------------------------------------------------------------------------------------
ERROR:homeassistant.bootstrap:Invalid config for [sensor.template]: [test INVALID sensor] is an invalid option for [sensor.template]. Check: sensor.template->sensors->test INVALID sensor.
======================================================================================== 1 failed, 6 passed, 1 pytest-warnings in 123.86 seconds =========================================================================================
````

The `false` return is coming from this code in `_setup_component`
````
                # Validate platform specific schema
                if hasattr(platform, 'PLATFORM_SCHEMA'):
                    try:
                        p_validated = platform.PLATFORM_SCHEMA(p_validated)
                    except vol.MultipleInvalid as ex:
                        _log_exception(ex, '{}.{}'.format(domain, p_name),
                                       p_validated)
                        return False
````